CMakeLists.txt

```
@@ -312,6 +316,21 @@ if(HPX_WITH_UNITY_BUILD)
   set(HPX_WITH_UNITY_BUILD_OPTION UNITY_BUILD)
 endif()
 
+# ##############################################################################
+# C++20 modules configuration
+# ##############################################################################
+hpx_option(
+  HPX_WITH_CXX_MODULES BOOL "Enable C++20 modules mode (default: OFF)." OFF
+  ADVANCED
+)
+if(HPX_WITH_CXX_MODULES)
+  hpx_info("HPX will build and install C++20 module interface units for downstream users.")
+  hpx_info("Note: HPX itself will still be built using traditional headers.")
+  add_compile_definitions(HPX_HAVE_CXX_MODULES)
```

Generally, all preprocessor macros are put into a special file (`<hpx/config/defines.hpp>`) instead of being added to the compiler's command line. This ensures that all targets that depend on HPX (even external ones) will see a consistent set of preprocessor definitions.
```suggestion
  hpx_add_config_define(HPX_HAVE_CXX_MODULES)
```